src/keysignature.ts

```diff
@@ -229,6 +229,30 @@ export class KeySignature extends StaveModifier {
     this.formatted = true;
   }
 
+  protected calculateWidth(): void {
```

Comment:
Should this also set `this.formatted = true`, so that if `this.getWidth` is called multiple times, it doesn't keep making `this.width` grow?

Reply:
We cannot set formatted to true. We cannot calculate the shifts without the stave. This is somehow a quick fix. The class deserves a refactoring.

Reply:
Ah, I see. Thanks for explaining.

Reply:
I also see this.width is being set to 0 at the beginning, so this doesn't seem like it'll cause this.width to grow incorrectly like I first thought.